File: output/deepseek_r1/avatar/algo_based_translation/python/java/codeforces_387_A.java

import java.util.Scanner;

public class codeforces_387_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        String[] time1Parts = scanner.nextLine().split(":");
        int h1 = Integer.parseInt(time1Parts[0]);
        int m1 = Integer.parseInt(time1Parts[1]);
        int total1 = h1 * 60 + m1;
        
        String[] time2Parts = scanner.nextLine().split(":");
        int h2 = Integer.parseInt(time2Parts[0]);
        int m2 = Integer.parseInt(time2Parts[1]);
        int total2 = h2 * 60 + m2;
        
        int diff = total1 - total2;
        diff %= 1440;
        if (diff < 0) {
            diff += 1440;
        }
        
        int hours = diff / 60;
        int minutes = diff % 60;
        
        System.out.printf("%02d:%02d%n", hours, minutes);
    }
}

File: output/deepseek_r1/avatar/algo_based_translation/java/python/successful_translation/codeforces_387_A.py

s = input().split(':')
t = input().split(':')
x_total = int(s[0]) * 60 + int(s[1])
y_total = int(t[0]) * 60 + int(t[1])
diff = (x_total - y_total) % 1440
print(f"{diff // 60:02d}:{diff % 60:02d}")

